### DISCUSSION TITLE
JumpServer to access cisco devices

### DISCUSSION THREAD

**Discussion by oscsalga** [carlmontanari/scrapli#293] [Q&A]
its weird to access devices with just one ssh connection, almost always we need to connect to a linux server in order from there to connect to the devices.

How can we do that???

**Comment by carlmontanari** [comment]
tl;dr - scrapli execs ssh on your system (with default system transport), so set it up however you want in your ssh config file. https://carlmontanari.github.io/scrapli/user_guide/basic_usage/#ssh-config-support

**Comment by oscsalga** [comment]
Hello, thanks for answering.
im trying with AsyncIOSXRDriver but i cannot manage to make it work, here is the dictionary im using:

IOSXR_DEVICE1 = {
    "host": "x.x.x.x",
    "auth_username": "x.x.x.x",
    "auth_password": "x.x.x.x",
    "auth_strict_key": False,
    "port": 22,
    "transport": "asyncssh",
    "driver": AsyncIOSXRDriver,
    "ssh_config_file": "/Users/oscsalga/ssh_configPrueba",
    
and here is the error:

socket.gaierror: [Errno 8] nodename nor servname provided, or not known

i tried with 

from scrapli.driver.core import IOSXEDriver

my_device = {
    "host": "172.18.0.11",
    "auth_username": "scrapli",
    "auth_password": "scrapli",
    "auth_strict_key": False,
    "ssh_config_file": "~/my_ssh_config",
}

with IOSXEDriver(**my_device) as conn:
    print(conn.get_prompt())
    
    and is working fine

**Reply by carlmontanari** [reply]
*system* transport uses standard ssh which is why sync transport works, no other transport does, they use their own implementations (i.e. asyncssh, ssh2). for asyncssh look -> [here](https://asyncssh.readthedocs.io/en/stable/api.html#config-file-support)
